interpreter/cling/include/cling/Interpreter/RuntimePrintValue.h

```diff
@@ -154,8 +154,8 @@ namespace cling {
     inline auto printValue_impl(
         const CollectionType* obj,
         typename std::enable_if<
-            std::is_reference<decltype(*(obj->begin()))>::value>::type* = 0)
-        -> decltype(++(obj->begin()), obj->end(), std::string()) {
+            std::is_reference<decltype(*std::begin(*obj))>::value>::type* = 0)
+        -> decltype(std::begin(*obj), std::end(*obj), std::string()) {
```

Comment:
As you don't check for `++` anymore, this simplifies to
```suggestion
            std::is_reference<decltype(*std::begin(*obj))>::value>::type* = 0)
        -> decltype(std::end(*obj), std::string()) {
```
(`std::begin(*obj)` is already checked in the `is_reference` part.)

Reply:
true, thanks!